packages/discord-bot/src/utils/stateEmbed.ts

```diff
@@ -0,0 +1,39 @@
+import { MessageEmbed, StringMappedInteractionTypes } from 'discord.js';
+import { UserState } from '../interfaces/UserState';
+
+const formatDate = (date: Date): string =>
+  `${date.getDate()}-${date.getMonth()}-${date.getFullYear()}`;
+
+export const getStateEmbed = (state: UserState): MessageEmbed => {
+  const embed = new MessageEmbed()
+    .setTitle(state.username)
+    .setDescription(
+      state.hasPraiseGiverRole && state.activated
+        ? 'Your account is activated and has praise powers.'
+        : state.hasPraiseGiverRole
+        ? "You have praise powers, however your account isn't activated(use `/activate` command)."
+        : "Your account is activated, however you don't have praise powers."
+    )
+    .setThumbnail(
+      `https://cdn.discordapp.com/avatars/${state.id}/${state.avatar || 'oops'}`
+    );
```

Comment:
why is the ` || oops` necessary? wouldn't state.avatar just be empty?

Also, if we need a conditional - should we make it explicit and only run `setThumbnail` if `state.avatar` is defined?